Added files via upload

lucas1richard opened the pull request:
This version expands the User Interface. I have not been able to connect my CrazyRadioPA in this version or the original version.

lucas1richard committed:
Added files via upload

ataffanel commented:
Hi,
I guess your intention was to do this PR in the bitcraze repos, there: https://github.com/bitcraze/crazyflie-chrome-client?

I tested the master branch and it works for me. What operating system are you using? On linux you will have to add a udev-rule file for Chrome to be allowed to access Crazyradio. By the way this commit does not work for me, I get the same error as you so something changed.

One comment on the commit: can you avoid commiting anything under `bower_components` and `Crazyflie` folders. Those files are generated by Bower, only the source files (currently at the root of the project) should be modified.

I see that you have added the gamepad VID/PID to the manifest, this is not required since we are accessing it with the Gamepad api (same as http://html5gamepad.com/), not the chrome-specific usb api.

lucas1richard commented on Crazyflie/init.js:
```diff
@@ -0,0 +1,168 @@
+      rollFactor:     30,
+      pitchFactor:    30,
+      yawFactor:      200,
+      thrustFactor:   5500,
```
Should be changed to 55000

lucas1richard commented:
Good afternoon Arnaud,

I wanted to stay away from the master as much as possible until I know my
version works well. I'm using Windows 10 with the latest version of Chrome
(Version 49.0.2623.87 m).

The VID/PID 6421/30583 were included in the original version and I didn't
change them. The 1915/7777 were added in accordance with the usb/radio
protocol provided on the bitcraze wiki. My Xbox 360 gamepad connects
perfectly to the Chrome client.

The Crazyradio works fine with the VM but the Xbox 360 controller always
shows busy with a previous request on the VM. I think that must be due to
Windows 10.

Best regards,
Richard

On Tue, Mar 29, 2016 at 5:18 AM Arnaud Taffanel notifications@github.com
wrote:

> Hi,
> I guess your intention was to do this PR in the bitcraze repos, there:
> https://github.com/bitcraze/crazyflie-chrome-client?
> 
> I tested the master branch and it works for me. What operating system are
> you using? On linux you will have to add a udev-rule file for Chrome to be
> allowed to access Crazyradio. By the way this commit does not work for me,
> I get the same error as you so something changed.
> 
> One comment on the commit: can you avoid commiting anything under
> bower_components and Crazyflie folders. Those files are generated by
> Bower, only the source files (currently at the root of the project) should
> be modified.
> 
> I see that you have added the gamepad VID/PID to the manifest, this is not
> required since we are accessing it with the Gamepad api (same as
> http://html5gamepad.com/), not the chrome-specific usb api.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/lucas1richard/crazyflie-chrome-client/pull/1#issuecomment-202795384

ataffanel commented:
Hi Richard,

You are right for VID/PID, now I wonder how this can work at all (I did that quite some time ago...). Anyway for me the master branch actually works but yours don't. I am on Linux though, I will test on windows during the day and report my results

Best,
/Arnaud

ataffanel commented:
Sorry I seems to be pretty confused on this one, it was a long time. So:
- For VID/PID the master branch is correct. 1915/7777 are hexadecimal but you cannot put hexadecimal numbers in a json so 6421/30583 is the same but in decimal.
- The master branch was not working on Windows, I changed it slightly to follow the way other projects write permissions and now it is working (see https://github.com/bitcraze/crazyflie-chrome-client/commit/63765dc013a188ba257cf9775cdd51bcefa83d9c). I tested with Chrome 49 on Windows 10.
- I tried the same change on your branch but it still does not work, though I am not getting the usb error anymore in the console.

Bower is just handling dependencies, not generating anything, I was confused in my previous message. However you should modify the code in the root instead of adding new code in the crazyflie folder: it would be easier to track the changes.

lucas1richard commented:
I'm not sure what's going either because the modification to the manifest
does not enable the extension to work on my laptop. I downloaded the
extension directly from github and added the jquery and underscore. Now the
extension cannot find the crazyradio dongle either. On my version, it still
finds the radio but there is still has an error claiming the interface.

Unfortunately I must return my Crazyflie 2.0 because there is an issue with
board. I will keep the Crazyradio PA for now assuming it does not have a
hardware issue.

On Wed, Mar 30, 2016 at 5:01 AM Arnaud Taffanel notifications@github.com
wrote:

> Sorry I seems to be pretty confused on this one, it was a long time. So:
> - For VID/PID the master branch is correct. 1915/7777 are hexadecimal
>   but you cannot put hexadecimal numbers in a json so 6421/30583 is the same
>   but in decimal.
> - The master branch was not working on Windows, I changed it slightly
>   to follow the way other projects write permissions and now it is working
>   (see bitcraze@63765dc
>   https://github.com/bitcraze/crazyflie-chrome-client/commit/63765dc013a188ba257cf9775cdd51bcefa83d9c).
>   I tested with Chrome 49 on Windows.
> - I tried the same change on your branch but it still does not work,
>   though I am not getting the usb error anymore in the console.
> 
> Bower is just handling dependencies, not generating anything, I was
> confused in my previous message. However you should modify the code in the
> root instead of adding new code in the crazyflie folder: it would be easier
> to track the changes.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/lucas1richard/crazyflie-chrome-client/pull/1#issuecomment-203333483

ataffanel commented:
This is too bad for you Crazyflie 2.0.

Have you installed the driver in Windows as described in the wiki: https://wiki.bitcraze.io/doc:crazyradio:install_windows_zadig

I tried to remove and reinstalled the driver on my windows and now everything stopped working so there is something going on with the driver.

If you try to install the driver and connect the master branch you should see a red LED both in the client and on the Crazyradio. If it is the case then it will work with a Crazyflie.
